scripts/userListPerRegion/users.py

```diff
@@ -0,0 +1,244 @@
+#!/usr/bin/env python
+# -- encoding: utf-8 --
+#
+# Copyright 2015-2016 Telefónica Investigación y Desarrollo, S.A.U
+#
+# This file is part of FI-WARE project.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+#
+# You may obtain a copy of the License at:
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For those usages not covered by the Apache version 2.0 License please
+# contact with opensource@tid.es
+#
+"""Get the users list of a specific FIWARE Lab region.
+
+Usage:
+  users --user=<username> --pass=<password> --region=<region> [--out=<filename>]
+  users -h | --help
+  users -v | --version
+
+Options:
+  -h --help           Show this screen.
+  -v --version        Show version.
+  -o --out=<filename> Store the information if the file <filename>.
+  --user=<username>   Admin user that request the data.
+  --pass=<password>   Admin password of the user.
+  --region=<region>   Region name that we want to recover the information.
+
+"""
+
+author = 'fla'
+
+from docopt import docopt
+import requests
+import os
+import json
+
+__version__ = '1.0.0'
+
+KEYSTONE_URL = 'http://cloud.lab.fiware.org:4730'
+API_V2 = 'v2.0'
+API_V3 = 'v3'
+TOKENS = 'tokens'
+CONTENT_TYPE = 'application/json'
+ACCEPT = CONTENT_TYPE
+ENDPOINT_GROUPS = 'OS-EP-FILTER/endpoint_groups'
+PROJECT_DETAILS = '%s/projects'
+ROLE_ASSIGNMENT = 'role_assignments?scope.project.id=%s'
+USER_DETAILS = 'users/%s'
+
+def gettoken(username, password):
+    """
+    Get the admin token of a corresponding region administrator.
+
+    :param username: the admin user name
+    :param password: the admin password
+    :return: the authenticated token
+    """
+    print('\nGetting administrator token...')
+
+    body = '''
+    {
+        "auth": {
+            "tenantName": "admin",
+            "passwordCredentials": {
+                "username": "%s",
+                "password": "%s"
+            }
+        }
+    }
+    ''' % (username, password)
+
+    url = os.path.join(KEYSTONE_URL, API_V2, TOKENS)
+    headers = {'content-type': CONTENT_TYPE, 'accept': ACCEPT}
+
+    response = requests.post(url=url, headers=headers, data=body)
+
+    info = json.loads(response.text)
+    token = info['access']['token']['id']
+
+    return token
+
+def getregionid(token, region):
+    """
+    Get the corresponding id of the region to check.
+
+    :param token: The admin token.
+    :param region: The region to get the id.
+    :return: the id of the region.
+    """
+    print('Getting id of the requested region...')
```

Comment:
There is a difference between what is expected the method do, attending the name and the description, and what the method really do.

The method name/description looks that is get the ID of the region (what is not very useful, because the region is already an ID)

But what the method really do is to look the region in the filter. And the method will fail if the user is not authorized in that region. Isn't it?